instance/tasks.py

```diff
@@ -127,3 +141,15 @@ def clean_up():
     """
     shut_down_obsolete_pr_sandboxes()
     terminate_obsolete_appservers_all_instances()
+
+
+@db_periodic_task(crontab())
+def reconfigure_dirty_load_balancers():
+    """
+    Any load balancers that are dirty need to be reconfigured.
+
+    This task runs every minute.
+    """
+    logger.info('Reconfiguring all dirty load balancers')
+    for load_balancer in LoadBalancingServer.objects.filter(configuration_state=LoadBalancingServer.DIRTY):
+        load_balancer.reconfigure()
```

Comment:
I think that if the program crashed or was control-c'd out of while a reconfiguration was going on, we'd be stuck in that state. Maybe we should just exclude `CLEAN` states instead? That wouldn't have an effect on the logic, and it would save us from these cases, but there would then guaranteed be at least 1 call to the DB per minute from this task (the one to perform the 'update'. We could maybe prevent that by first checking to see if the state is `IN_PROGRESS` or `IN_PROGRESS_AND_DIRTY` and kick the thread out early to prevent that DB call.

Reply:
If an LB is stuck in `IN_PROGRESS` state, calling the `reconfigure()` method won't help, since it won't do anything except for marking the LB as in progress and dirty, but we would still be stuck in that state; see my comment above.

Reply:
Yeah, not sure what I was thinking when I commented that. :)